lib/index.js

```diff
@@ -1,54 +1,107 @@
-exports.RevoluteJoint = require('./joint/RevoluteJoint');
-exports.RopeJoint = require('./joint/RopeJoint');
-exports.WeldJoint = require('./joint/WeldJoint');
-exports.WheelJoint = require('./joint/WheelJoint');
-
-exports.Settings = require('./Settings');
-
-exports.internal.Sweep = require('./common/Sweep');
-exports.internal.stats = require('./common/stats'); // todo: remove this
-exports.internal.Manifold = require('./Manifold');
-exports.internal.Distance = require('./collision/Distance');
-exports.internal.TimeOfImpact = require('./collision/TimeOfImpact');
-exports.internal.DynamicTree = require('./collision/DynamicTree');
-exports.internal.Settings = exports.Settings;
+import Serializer from './serializer/index';
+
+import Math from './common/Math';
+import Vec2 from './common/Vec2';
+import Vec3 from './common/Vec3';
+import Mat22 from './common/Mat22';
+import Mat33 from './common/Mat33';
+import Transform from './common/Transform';
+import Rot from './common/Rot';
+
+import AABB from './collision/AABB';
+
+import Shape from './Shape';
+import Fixture from './Fixture';
+import Body from './Body';
+import Contact from './Contact';
+import Joint from './Joint';
+import World from './World';
+
+import Circle from './shape/CircleShape';
+import Edge from './shape/EdgeShape';
+import Polygon from './shape/PolygonShape';
+import Chain from './shape/ChainShape';
+import Box from './shape/BoxShape';
+
+import { CollideCircles } from './shape/CollideCircle';
+import { CollideEdgeCircle } from './shape/CollideEdgeCircle';
+import { CollidePolygons } from './shape/CollidePolygon';
+import { CollidePolygonCircle } from './shape/CollideCirclePolygone';
+import { CollideEdgePolygon } from './shape/CollideEdgePolygon';
+
+import DistanceJoint from './joint/DistanceJoint';
+import FrictionJoint from './joint/FrictionJoint';
+import GearJoint from './joint/GearJoint';
+import MotorJoint from './joint/MotorJoint';
+import MouseJoint from './joint/MouseJoint';
+import PrismaticJoint from './joint/PrismaticJoint';
+import PulleyJoint from './joint/PulleyJoint';
+import RevoluteJoint from './joint/RevoluteJoint';
+import RopeJoint from './joint/RopeJoint';
+import WeldJoint from './joint/WeldJoint';
+import WheelJoint from './joint/WheelJoint';
+
+import Settings from './Settings';
+
+import Sweep from './common/Sweep';
+import Manifold from './Manifold';
+import Distance from './collision/Distance';
+import TimeOfImpact from './collision/TimeOfImpact';
+import DynamicTree from './collision/DynamicTree';
+
+export default {
+  Serializer,
+
+  Math,
+  Vec2,
+  Vec3,
+  Mat22,
+  Mat33,
+  Transform,
+  Rot,
+
+  AABB,
+
+  Shape,
+  Fixture,
+  Body,
+  Contact,
+  Joint,
+  World,
+
+  Circle,
+  Edge,
+  Polygon,
+  Chain,
+  Box,
+
+  DistanceJoint,
+  FrictionJoint,
+  GearJoint,
+  MotorJoint,
+  MouseJoint,
+  PrismaticJoint,
+  PulleyJoint,
+  RevoluteJoint,
+  RopeJoint,
+  WeldJoint,
+  WheelJoint,
+
+  Settings,
+};
```

Comment:
Maybe instead of exporting one object containing all modules we could also export every module by its own. This would work by `export { default as Math} from "..."`. Exporting all modules in one object may infer with tree shaking, but I haven't tested it, so i'm not sure.

On the other hand doing it this way better separates these modules from the internal modules.

Reply:
I will try it.